src/erlcloud_cloudformation.erl

```diff
@@ -0,0 +1,353 @@
+
+-spec describe_stacks(params(), aws_config()) -> {ok, cloudformation_list()}.
+describe_stacks(Params, Config = #aws_config{}) ->
+    {ok, XmlNode} = cloudformation_request(Config, "DescribeStacks", Params),
+
+    NextToken = erlcloud_xml:get_text("/DescribeStacksResponse/DescribeStacksResult/NextToken", XmlNode, undefined),
+
+    [{next_token, NextToken}, extract_described_stacks_result(xmerl_xpath:string("/DescribeStacksResponse/DescribeStacksResult", XmlNode))].
+
+-spec get_stack_policy(params(), aws_config()) -> {ok, cloudformation_list()}.
+get_stack_policy(Params, Config = #aws_config{}) ->
+    {ok, XmlNode} = cloudformation_request(Config, "GetStackPolicy", Params),
+
+    extract_stack_policy_body(xmerl_xpath:string("/GetStackPolicyResponse/GetStackPolicyResult", XmlNode)).
+
+
+-spec describe_stack_events_all(params(), aws_config()) -> {ok, cloudformation_list()}.
+describe_stack_events_all(Params, Config = #aws_config{}) ->
+    list_all(fun describe_stack_events/2, Params, Config, []).
+
+-spec describe_stack_events(params(), aws_config()) -> {ok, cloudformation_list()}.
+describe_stack_events(Params, Config = #aws_config{}) ->
+    {ok, XmlNode} = cloudformation_request(Config, "DescribeStackEvents", Params),
+
+    NextToken = erlcloud_xml:get_text("/DescribeStackEventsResponse/DescribeStackEventsResult/NextToken", XmlNode, undefined),
+
+    [{next_token, NextToken}, extract_described_stack_events_result(xmerl_xpath:string("/DescribeStackEventsResponse/DescribeStackEventsResult", XmlNode))].
+
+-spec get_template(string(), aws_config()) -> {ok, cloudformation_list()}.
+get_template(StackName, Config = #aws_config{}) ->
+    {ok, XmlNode} = cloudformation_request(Config, "GetTemplate", [{"StackName", StackName}]),
+    extract_template_response(XmlNode).
+
+-spec get_template_summary(params(), aws_config()) -> {ok, cloudformation_list()}.
+get_template_summary(Params, Config = #aws_config{}) ->
+
+    {ok, XmlNodes} = cloudformation_request(Config, "GetTemplateSummary", Params),
+    extract_template_summary_response(XmlNodes).
+
+-spec describe_account_limits_all(aws_config()) -> {ok, cloudformation_list()}.
+describe_account_limits_all(Config = #aws_config{}) ->
+    list_all(fun describe_account_limits/2, [], Config, []).
+
+-spec describe_account_limits(aws_config(), string()) -> {ok, cloudformation_list()}.
+describe_account_limits(Params, Config = #aws_config{}) ->
+    {ok, XmlNodes} = cloudformation_request(Config, "DescribeAccountLimits", Params),
+    NextToken = erlcloud_xml:get_text("/DescribeAccountLimitsResponse/DescribeAccountLimitsResult/NextToken", XmlNodes, undefined),
+    [{next_token, NextToken},  extract_accout_limits_response(xmerl_xpath:string("/DescribeAccountLimitsResponse", XmlNodes))].
+
+%%==============================================================================
+%% Internal functions
+%%==============================================================================
+
```

Comment:
I am aware that some of the names for these extraction functions are a bit 'flakey' I am open to suggestions about better naming conventions for these. 

The problem I had here was that for each XML 'level' you need a different function for extraction